roles/matrix-bot-go-neb/templates/config.yaml.j2

```diff
@@ -0,0 +1,44 @@
+# Go-NEB Configuration File
+#
+# This file provides an alternative way to configure Go-NEB which does not involve HTTP APIs.
+#
+# This file can be supplied to go-neb by the environment variable `CONFIG_FILE=config.yaml`.
+# It will force Go-NEB to operate in "config" mode. This means:
+#   - Go-NEB will ONLY use the data contained inside this file.
+#   - All of Go-NEB's /admin HTTP listeners will be disabled. You will be unable to add new services at runtime.
+#   - The environment variable `DATABASE_URL` will be ignored and an in-memory database will be used instead.
+#
+# This file is broken down into 4 sections which matches the following HTTP APIs:
+#   - /configureClient
+#   - /configureAuthRealm
+#   - /configureService
+#   - /requestAuthSession (redirects not supported)
+
+# The list of clients which Go-NEB is aware of.
+# Delete or modify this list as appropriate.
+# See the docs for /configureClient for the full list of options:
+# https://matrix-org.github.io/go-neb/pkg/github.com/matrix-org/go-neb/api/index.html#ClientConfig
+clients:
+  {{ matrix_bot_go_neb_clients }}
```

Comment:
Shouldn't we use `|to_json` here and in other instances below?

Reply:
Hi! Thank you for the feedback!

The `{{ matrix_bot_go_neb_clients }}` contains the entire code block from the vars.yaml file.

For example:
```
matrix_bot_go_neb_clients: 
  - UserID: "@goneb:localhost"
    AccessToken: "MDASDASJDIASDJASDAFGFRGER"
    DeviceID: "DEVICE1"
    HomeserverURL: "http://localhost:8008"
    Sync: true
    AutoJoinRooms: true
    DisplayName: "Go-NEB!"
    AcceptVerificationFromUsers: [":localhost:8008"]
```

Then that block gets filled in as-is in the config.yaml from the bot. Like the [sample.config.yaml](https://github.com/matrix-org/go-neb/blob/master/config.sample.yaml#L21) from the upstream repo.

Adding a |to_json will break the bot, because it's expecting yaml.

Do you prefer a custom array/dict in the vars.yaml which we then can loop with with_items?
And then in the config.yaml.j2 something like:
```
clients:
    {% for client in matrix_bot_go_neb_clients %}
  - UserID: {{ client.user_id|to_json }}
    AccessToken: {{ client.access_token|to_json }}
    DeviceID: {{ client.device_id|to_json }}
etc
etc
    {% endfor %}
```

Reply:
It reminds me of `matrix_appservice_irc_ircService_servers` that we use in the `matrix-appservice-irc` role.

Ansible parses that value into a native list + dict, and then you can use `|to_json` to serialize it as JSON (well, YAML).

Are your values being turned into a string somehow?

Reply:
Ah yes, it does look a lot like `matrix_appservice_irc_ircService_servers`. I might've misunderstood the workings of `|to_json`.

I will do some testing!

Reply:
Tested with |to_json and everything is working. So it was just a misunderstanding of that function on my part. Corrected in https://github.com/spantaleev/matrix-docker-ansible-deploy/pull/929/commits/49028f1b0588bf5aefdd2985e5749958b043b4b7